src/attached_probe.cpp

```diff
@@ -1124,10 +1130,28 @@ void AttachedProbe::attach_watchpoint(int pid, const std::string& mode)
 
   for (int cpu : cpus)
   {
-    int perf_event_fd = bpf_attach_perf_event_raw(
-        progfd_, &attr, pid, cpu, -1, 0);
+    // We copy paste the code from bcc's bpf_attach_perf_event_raw here
+    // because we need to know the exact error codes (and also we don't
+    // want bcc's noisy error messages).
+    int perf_event_fd = syscall(
+        __NR_perf_event_open, &attr, pid, cpu, -1, PERF_FLAG_FD_CLOEXEC);
     if (perf_event_fd < 0)
+    {
+      if (errno == ENOSPC)
+        throw EnospcException("No more HW registers left");
+      else
+        throw std::runtime_error("Error attaching probe: " + probe_.name);
```

Comment:
`std::system_error(errno, std::generic_category(), "error attaching probe"...);`

to maintain the err code